File: Arcade/TheCore/Java/IsSumOfConsecutive2.java
final class IsSumOfConsecutive2 {
	
	int isSumOfConsecutive2(int n) {
	    int ways = 0;
	    
	    for (int i = 1; i <= n; i++) {
	        int sum = i;
	        
	        for (int j = i + 1; j < n; j++) {
	            if ((sum += j) > n) break;
	            else if (sum == n) ways++;
	        }
	    }
	    
	    return ways;
	}
}

File: Arcade/TheCore/Javascript/isSumOfConsecutive2.js
function isSumOfConsecutive2(n) {
    let ways = 0;
    
    for (let i = 1; i <= n; i++) {
        let sum = i;
        
        for (let j = i + 1; j <= n; j++) {
            if ((sum += j) > n) break
            else if (sum == n) ways++
        }
    }
    
    return ways;
}